cluster/examples/kubernetes/ceph/create-external-cluster-resources.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# this script creates all the users/keys on the external cluster
+# those keys will be injected via the import-external-cluster.sh once this one is done running
+# so you can run import-external-cluster.sh right after this script
+set -Eeuo pipefail
+
+#############
+# VARIABLES #
+#############
+
+: "${CLIENT_CHECKER_NAME:=client.healthchecker}"
+
+#############
+# FUNCTIONS #
+#############
+
+function is_available {
+  command -v "$@" &>/dev/null
+}
+
+function checkEnv() {
+  if ! is_available ceph; then
+    echo "'ceph' binary is expected'"
+    exit 1
+  fi
+
+  if ! ceph -s 1>/dev/null; then
+    echo "cannot connect to the ceph cluster"
+    exit 1
+fi
+}
+
+function createCheckerKey() {
+  checkerKey=$(ceph auth get-or-create "$CLIENT_CHECKER_NAME" mon 'allow r, allow command quorum_status'|awk '/key =/ { print $3}')
+  echo "export ROOK_EXTERNAL_CHECKER_SECRET=$checkerKey"
+  echo "export ROOK_EXTERNAL_CHECKER_USERNAME=$CLIENT_CHECKER_NAME"
+}
+
+function createCephCSIKeyringRBDNode() {
+  cephCSIKeyringRBDNodeKey=$(ceph auth get-or-create client.csi-rbd-node mon 'profile rbd' osd 'profile rbd'|awk '/key =/ { print $3}')
+  echo "export CSI_RBD_NODE_SECRET_SECRET=$cephCSIKeyringRBDNodeKey"
+}
+
+function createCephCSIKeyringRBDProvisioner() {
+  cephCSIKeyringRBDProvisionerKey=$(ceph auth get-or-create client.csi-rbd-provisioner mon 'profile rbd' mgr 'allow rw' osd 'profile rbd'|awk '/key =/ { print $3}')
+  echo "export CSI_RBD_PROVISIONER_SECRET=$cephCSIKeyringRBDProvisionerKey"
+}
+
+function createCephCSIKeyringCephFSNode() {
+  cephCSIKeyringCephFSNodeKey=$(ceph auth get-or-create client.csi-cephfs-node mon 'allow r' mgr 'allow rw' osd 'allow rw tag cephfs *=*' mds 'allow rw'|awk '/key =/ { print $3}')
+  echo "export CSI_CEPHFS_NODE_SECRET=$cephCSIKeyringCephFSNodeKey"
+}
+
+function createCephCSIKeyringCephFSProvisioner() {
+  cephCSIKeyringCephFSProvisionerKey=$(ceph auth get-or-create client.csi-cephfs-provisioner mon 'allow r' mgr 'allow rw' osd 'allow rw tag cephfs metadata=*'|awk '/key =/ { print $3}')
+  echo "export CSI_CEPHFS_PROVISIONER_SECRET=$cephCSIKeyringCephFSProvisionerKey"
+}
+
```

Comment:
For the auth keys to adopt least privilege shouldn't they be restricted to specific pools. For example, ceph cluster admin creates poola and poolb, and the keys for rook ceph would be limited to those pools..Isn't that what would be needed for multi-tenancy?

Reply:
This is meant to be an example but yes we could restrict to a specific pool. Once the key is injected (as a secret in Rook) the caps (Ceph side) can always be changed and adapted.